CMakeLists.txt

```diff
@@ -3,32 +3,43 @@
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
 cmake_minimum_required(VERSION 3.14.0 FATAL_ERROR)
-project(
-    umf
-    VERSION 0.1.0
-    LANGUAGES C)
-
 # needed when UMF is used as an external project
 set(UMF_CMAKE_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 
 list(APPEND CMAKE_MODULE_PATH "${UMF_CMAKE_SOURCE_DIR}/cmake")
-include(${UMF_CMAKE_SOURCE_DIR}/cmake/helpers.cmake)
+include(helpers)
+
+# Establish version vars based on the git tag and what 'git describe' reports.
+# If there's no git available in the OS, the version will be set to "0.0.0".
+# Otherwise we'll use 3-component version: major.minor.patch, just for CMake's
+# sake. A few extra variables will be set for Win dll metadata.
+#
+# For all other usages (beside CMake and Win dll), we'll be using semver aligned
+# version "UMF_VERSION", which is in line with our tags (e.g. "1.5.0-rc2").
+set_version_variables()
+project(
+    umf
+    VERSION ${UMF_CMAKE_VERSION}
+    LANGUAGES C)
+# BUGFIX = "true" if Patch version > 0
+if(CMAKE_PROJECT_VERSION_PATCH GREATER 0)
+    set(UMF_VERSION_BUGFIX TRUE)
+endif()
+
+message(
+    STATUS
+        "UMF CMake version: ${UMF_CMAKE_VERSION} (via CMake: ${CMAKE_PROJECT_VERSION})\n"
+        "UMF version: ${UMF_VERSION}\n"
```

Comment:
Please print each line in a separate `message(STATUS ...` because now it looks like:
```
-- GIT_VERSION: v0.1.0-28-g4ab61a7
-- UMF CMake version: 0.1.0 (via CMake: 0.1.0)
UMF version: 0.1.0-git28.g4ab61a7
UMF_VERSION_REVISION: 28
UMF_VERSION_PRERELEASE: FALSE
UMF_VERSION_PRIVATE: TRUE
UMF_VERSION_BUGFIX: FALSE
```

Reply:
we shouldn't care - these messages are temporary, for debugging purpose, to check if all vars are set properly.